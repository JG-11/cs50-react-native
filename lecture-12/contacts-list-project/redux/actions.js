import { retrieveQuote } from './../API'

export const UPDATE_USER = 'UPDATE_USER'
export const UPDATE_CONTACT = 'UPDATE_CONTACT'
export const GET_QUOTE = 'GET_QUOTE'

export const updateUser = update => ({
    type: UPDATE_USER,
    payload: update
})
export const addContact = newContact => ({
    type: UPDATE_CONTACT,
    payload: newContact
})
export const getQuote = () => async dispatch => {
    const quote = await retrieveQuote()

    dispatch({ type: GET_QUOTE, payload: quote })
}
