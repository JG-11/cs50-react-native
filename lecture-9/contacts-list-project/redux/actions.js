export const UPDATE_USER = 'UPDATE_USER'
export const UPDATE_CONTACT = 'UPDATE_CONTACT'

export const updateUser = update => ({
    type: UPDATE_USER,
    payload: update
})
export const addContact = newContact => ({
    type: UPDATE_CONTACT,
    payload: newContact
})