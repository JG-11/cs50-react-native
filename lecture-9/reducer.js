/*const merge = (prev, next) => Object.assign({}, prev, next)

const reducer = (state, update) => merge(state, update)*/

const reducer = (state, update) => ({
    ...state,
    ...update
})

let state = {}
state = reducer(state, {foo: 'foo'})
state = reducer(state, {bar: 'bar'})

console.log(state)