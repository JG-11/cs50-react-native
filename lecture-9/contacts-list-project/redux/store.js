import { createStore } from 'redux'
import reducer from './reducer'
import { addContact } from './actions'

const store = createStore(reducer)

/*
store.dispatch(updateUser({foo: 'foo'}))
store.dispatch(updateUser({bar: 'bar'}))
*/

store.dispatch(addContact({name: 'Genaro', phone: '12121212'}))
store.dispatch(addContact({name: 'Leslie', phone: '1212323'}))
store.dispatch(addContact({name: 'Genaro', phone: '121212'}))

export default store