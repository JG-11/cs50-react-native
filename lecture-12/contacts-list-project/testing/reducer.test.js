import reducer from './../redux/reducer'
import * as actions from './../redux/actions'

const DEFAULT_STATE = {
  user: {},
  contacts: []
}

describe('contacts reducer', () => {
  it('successfully adds new user', () => {
    expect(reducer(DEFAULT_STATE, actions.addContact({
      name: 'Test user',
      phone: '123456'
    }))).toMatchSnapshot()
  })
})

describe('user reducer', () => {
  it('successfully updates user', () => {
    expect(reducer(DEFAULT_STATE, actions.updateUser({
      name: 'test user'
    }))).toMatchSnapshot()
  })
})
