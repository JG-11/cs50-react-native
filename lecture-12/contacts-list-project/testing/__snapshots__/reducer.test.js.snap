// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contacts reducer successfully adds new user 1`] = `
Object {
  "contacts": Array [
    Object {
      "name": "Test user",
      "phone": "123456",
    },
  ],
  "user": Object {
    "prevContact": Object {
      "name": "Test user",
      "phone": "123456",
    },
  },
}
`;

exports[`user reducer successfully updates user 1`] = `
Object {
  "contacts": Array [],
  "user": Object {
    "name": "test user",
  },
}
`;
